File: idn-ideation-system/frontend/src/types/project.ts
import type { NarrativeFormat, StoryEdge, StoryNode } from './graph'
import type { AIMessage } from './ai'

export interface ProjectMeta {
  id: string
  title: string
  theme: string
  format: NarrativeFormat
  created_at: string
  updated_at: string
  node_count: number
}

export interface Project {
  id: string
  title: string
  theme: string
  format: NarrativeFormat
  created_at: string
  updated_at: string
  nodes: StoryNode[]
  edges: StoryEdge[]
  chat_history: AIMessage[]
}

File: idn-ideation-system/backend/models/project.py
from datetime import datetime
from typing import Literal
from pydantic import BaseModel, Field
from .graph import StoryNode, StoryEdge


class ProjectMeta(BaseModel):
    id: str
    title: str
    theme: str = ""
    format: Literal["branching", "hypertext"] = "branching"
    created_at: datetime
    updated_at: datetime
    node_count: int = 0


class Project(BaseModel):
    id: str
    title: str
    theme: str = ""
    format: Literal["branching", "hypertext"] = "branching"
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    nodes: list[StoryNode] = []
    edges: list[StoryEdge] = []
    chat_history: list[dict] = []


class CreateProjectRequest(BaseModel):
    title: str
    theme: str = ""
    format: Literal["branching", "hypertext"] = "branching"


class SaveProjectRequest(BaseModel):
    title: str
    theme: str = ""
    format: Literal["branching", "hypertext"] = "branching"
    nodes: list[StoryNode] = []
    edges: list[StoryEdge] = []
    chat_history: list[dict] = []
